added SUSE and openSUSE

msmeissn opened the pull request:
Added the SUSE and openSUSE ecosystem.

https://github.com/ossf/osv-schema/issues/259

oliverchang commented on docs/schema.md:
```diff
@@ -387,6 +387,17 @@ The defined database prefixes and their "home" databases are:
+          <li>OSV Formatted URL: <code>https://ftp.suse.com/pub/projects/security/osv/</code></li>
+        </ul>
+      </td>
+    </tr>
```
Would you be able to add some ecosystem definitions here: https://github.com/ossf/osv-schema/blob/516347895e3541e04db8166f9c83a22250f5b44d/docs/schema.md#affectedpackage-field also ?

There's a few other Linux distros in there that could potentially be useful to base off.

msmeissn replied:
I added stuff there

oliverchang reviewed (commented):
awesome, thank you!

andrewpollock commented on validation/schema.json:
```diff
@@ -300,7 +300,7 @@
-      "pattern": "^(AlmaLinux|Alpine|Android|Bioconductor|Bitnami|ConanCenter|CRAN|crates.io|Debian|GHC|GitHub Actions|GIT|Go|Hackage|Hex|Linux|Maven|npm|NuGet|OSS-Fuzz|Packagist|Photon OS|Pub|PyPI|Red Hat|Rocky Linux|RubyGems|SwiftURL|Ubuntu)(:[[:digit:]]+)?"
+      "pattern": "^(AlmaLinux|Alpine|Android|Bioconductor|Bitnami|ConanCenter|CRAN|crates.io|Debian|GHC|GitHub Actions|GIT|Go|Hackage|Hex|Linux|Maven|npm|NuGet|openSUSE|OSS-Fuzz|Packagist|Photon OS|Pub|PyPI|Red Hat|Rocky Linux|RubyGems|SUSE|SwiftURL|Ubuntu)(:[[:digit:]]+)?"
     },
     "prefix": {
```
The prefix(es) (`SUSE-SU`?) need(s) to be added to the validation here.

msmeissn replied:
not in the expression below?

msmeissn replied:
i added in the expression below

andrewpollock commented on docs/schema.md:
```diff
@@ -387,6 +387,17 @@ The defined database prefixes and their "home" databases are:
       </td>
     </tr>
+    <tr>
+      <td><code>SUSE and openSUSE</code></td>
```
This needs to be the record prefix, which from a brief eyeball of https://lists.suse.com/pipermail/sle-security-updates/ should be `SUSE-SU`?

```suggestion
      <td><code>SUSE-SU</code></td>
```

msmeissn replied:
changed, with and not in code

andrewpollock commented on docs/schema.md:
```diff
@@ -685,13 +696,15 @@ The defined ecosystems are:
 | `npm` | The NPM ecosystem; the `name` field is an NPM package name.  |
 | `NuGet` | The NuGet package ecosystem. The `name` field is a NuGet package name.  |
 | `OSS-Fuzz` | For reports from the OSS-Fuzz project that have no more appropriate ecosystem; the `name` field is the name assigned by the OSS-Fuzz project, as recorded in the submitted fuzzing configuration.  |
+| `openSUSE` | The openSUSE ecosystem; The ecosystem specifier is the marketing name of the openSUSE distribution. With the `:` removed, it matches the value in the /etc/os-release `PRETTY_NAME` field. The `name` field is the name of the source RPM and accompanied by a purl. There is a `ecosystem_specific` specific array `binaries` of the associated RPM binary packages in this specific openSUSE d ...
```
Minor grammatical and styling:

```suggestion
| `openSUSE` | The openSUSE ecosystem; The ecosystem specifier is the marketing name of the openSUSE distribution. With the `:` removed, it matches the value in the `/etc/os-release` `PRETTY_NAME` field. The `name` field is the name of the source RPM and accompanied by a purl. There is an `ecosystem_specific` specific array `binaries` of the associated RPM binary packages in this specific openSUSE distribution. The ECOSYSTEM version ordering is the RPM versioncompare ordering, and the database uses the `introduced` and `fixed` boundaries.|
```

andrewpollock reviewed (approved):
LGTM with some minor style and grammar

oliverchang commented on docs/schema.md:
```diff
@@ -685,13 +696,15 @@ The defined ecosystems are:
 | `npm` | The NPM ecosystem; the `name` field is an NPM package name.  |
 | `NuGet` | The NuGet package ecosystem. The `name` field is a NuGet package name.  |
 | `OSS-Fuzz` | For reports from the OSS-Fuzz project that have no more appropriate ecosystem; the `name` field is the name assigned by the OSS-Fuzz project, as recorded in the submitted fuzzing configuration.  |
+| `openSUSE` | The openSUSE ecosystem; The ecosystem specifier is the marketing name of the openSUSE distribution. With the `:` removed, it matches the value in the /etc/os-release `PRETTY_NAME` field. The `name` field is the name of the source RPM and accompanied by a purl. There is a `ecosystem_specific` specific array `binaries` of the associated RPM binary packages in this specific openSUSE d ...
```
```suggestion
| `openSUSE` | The openSUSE ecosystem; The ecosystem string has a `:<RELEASE>` suffix representing the marketing name of the openSUSE distribution. `<RELEASE>` matches the value in the /etc/os-release `PRETTY_NAME` field. The `name` field is the name of the source RPM and accompanied by a purl. There is a `ecosystem_specific` specific array `binaries` of the associated RPM binary packages in this specific openSUSE distribution. The ECOSYSTEM version ordering is the RPM versioncompare ordering, and the database uses the `introduced` and `fixed` boundaries.|
```

oliverchang replied:
(One more suggestion for consistency with other ecosystem definitions)

oliverchang commented on docs/schema.md:
```diff
@@ -685,13 +696,15 @@ The defined ecosystems are:
 | `Red Hat` | The Red Hat package ecosystem; the `name` field is the name of a binary or source RPM. The ecosystem string has a `:<CPE>` suffix to scope the RPM to a specific Red Hat product stream. `<CPE>` is a translation of a Red Hat [Common Platform Enumerations](https://cpe.mitre.org/) (CPE) with the `cpe/:[oa]:(redhat):` prefix removed (for example, `Red Hat:rhel_aus:8.4::appstream` transla ...
 | `Rocky Linux` | The Rocky Linux package ecosystem; the `name` is the name of the source package. The ecosystem string might optionally have a `:<RELEASE>` suffix to scope the package to a particular Rocky Linux release. `<RELEASE>` is a numeric version.
 | `RubyGems` | The RubyGems ecosystem; the `name` field is a gem name.  |
+| `SUSE` | The SUSE ecosystem; The ecosystem specifier is the marketing name of the SUSE product. With the `:` removed, it matches the value in the /etc/os-release `PRETTY_NAME` field. The `name` field is the name of the source RPM and accompanied by a purl. There is a `ecosystem_specific` specific array `binaries` of the associated RPM binary packages in this specific SUSE product. The ECOSYSTEM ...
```
```suggestion
| `SUSE` | The SUSE ecosystem; The ecosystem string has a `:<RELEASE>` suffix representing the marketing name of the SUSE product. `<RELEASE>` matches the value in the /etc/os-release `PRETTY_NAME` field. The `name` field is the name of the source RPM and accompanied by a purl. There is a `ecosystem_specific` specific array `binaries` of the associated RPM binary packages in this specific SUSE product. The ECOSYSTEM version ordering is the RPM versioncompare ordering, and the database uses the `introduced` and `fixed` boundaries.|
```

oliverchang reviewed (commented):
thanks!

msmeissn commented:
Thanks. I had to resolve conflicts with the last ubuntu change, and squashed.

I also changed the wording again a bit, as we have "SUSE:Linux Enterprise Server" ... so the prefix includes part of the marketing name.

oliverchang commented on docs/schema.md:
```diff
@@ -696,13 +707,15 @@ The defined ecosystems are:
 | `npm` | The NPM ecosystem; the `name` field is an NPM package name.  |
 | `NuGet` | The NuGet package ecosystem. The `name` field is a NuGet package name.  |
 | `OSS-Fuzz` | For reports from the OSS-Fuzz project that have no more appropriate ecosystem; the `name` field is the name assigned by the OSS-Fuzz project, as recorded in the submitted fuzzing configuration.  |
+| `openSUSE` | The openSUSE ecosystem; The ecosystem specifier is the marketing name of the openSUSE distribution. With the `:` removed, it matches the value in the `/etc/os-release` `PRETTY_NAME` field. The `name` field is the name of the source RPM and accompanied by a purl. There is an `ecosystem_specific` specific array `binaries` of the associated RPM binary packages in this specific openSUS ...
```
Would you be able to make this consistent with the text for `SUSE` ? i.e. "The ecosystem string has a `:<RELEASE>` suffix representing the marketing name of the openSUSE distribution..."

msmeissn replied:
done so

oliverchang reviewed (commented):
thanks! Just one last comment.
